File: Electronics_Shop_17.Model/Requests/AddCartItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Electronics_Shop_17.Model.Requests
{
    public class AddCartItem
    {
        public int Quantity { get; set; }
        public int ProductId { get; set; }
        public int ShoppingCartId { get; set; }
        public int ProductColorId { get; set; }

    }
}

File: angular17_web/src/app/models/cart-item/add.cart.item.ts
export class AddCartItem {
  quantity!: number;
  productId!: number;
  shoppingCartId!: number;
  productColorId!: number;

  constructor(data?: any) {
    if (!data) return;

    this.quantity = data.quantity;
    this.productId = data.productId;
    this.shoppingCartId = data.shoppingCartId;
    this.productColorId = data.productColorId;
  }
}
